src/Parser/TypeParser.php

```diff
@@ -221,12 +221,15 @@ private function tryParseArray(TokenIterator $tokens, Ast\Type\TypeNode $type):
 	private function parseArrayShape(TokenIterator $tokens, Ast\Type\TypeNode $type): Ast\Type\TypeNode
 	{
 		$tokens->consumeTokenType(Lexer::TOKEN_OPEN_CURLY_BRACKET);
+		$tokens->tryConsumeTokenType(Lexer::TOKEN_PHPDOC_EOL);
 		$items = [$this->parseArrayShapeItem($tokens)];
 
 		while ($tokens->tryConsumeTokenType(Lexer::TOKEN_COMMA)) {
+			$tokens->tryConsumeTokenType(Lexer::TOKEN_PHPDOC_EOL);
 			$items[] = $this->parseArrayShapeItem($tokens);
 		}
 
+		$tokens->tryConsumeTokenType(Lexer::TOKEN_PHPDOC_EOL);
 		$tokens->consumeTokenType(Lexer::TOKEN_CLOSE_CURLY_BRACKET);
```

Comment:
I think that we should also allow line breaks before the comma, so that this works:

```
array{
    int
    , int
    , int
}
```

We might want to allow trailing commas as well:


```
array{
    int,
    int,
    int,
}
```

Reply:
I agree. Trailing commas are awesome :)

Reply:
Each of these examples should have a separate test case.

Reply:
Line breaks before commas 1f26308